hook!-full-game/src/singletons/Events.gd

```diff
@@ -6,3 +6,6 @@ signal player_state_changed(state)
 
 signal zero_gravity_area_entered()
 signal zero_gravity_area_exited()
+
+signal player_died
```

Comment:
You have this global signal for any node to react to the player's death and respawn.